sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnVectorUtils.java

```diff
@@ -59,19 +62,44 @@ public static Object toPrimitiveJavaArray(ColumnVector.Array array) {
 
   private static void appendValue(ColumnVector dst, DataType t, Object o) {
     if (o == null) {
-      dst.appendNull();
+      if (t instanceof CalendarIntervalType) {
+        dst.appendStruct(true);
+      } else {
+        dst.appendNull();
+      }
     } else {
-      if (t == DataTypes.ByteType) {
-        dst.appendByte(((Byte)o).byteValue());
+      if (t == DataTypes.BooleanType) {
```

Comment:
No of the functions in this file is perf critical. That's largely why I didn't put them in the ColumnVector class. This needs to be codegened whenever we need to do row -> columnarbatch conversion.